src/pytorch_lightning/utilities/data.py

```diff
@@ -371,18 +422,31 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
         )
         param_names = param_names[: len(args)]
 
-        if not hasattr(obj, "__pl_dl_args"):
-            obj.__pl_dl_args = args
-            obj.__pl_dl_kwargs = kwargs
-            obj.__pl_dl_arg_names = param_names
+        default_kwargs = {
+            param.name: param.default
+            for param in params.values()
+            if param.name != "self"
+            and param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)
+            and param.default != param.empty
+            and (param.name not in kwargs and param.name not in param_names)
+        }
+
+        kwargs = {**kwargs, **default_kwargs}
+
+        if not hasattr(obj, "__pl_saved_args"):
+            obj.__pl_saved_args = args
+            obj.__pl_saved_kwargs = kwargs
+            obj.__pl_saved_arg_names = param_names
 
-        # We want to use the latest possible value for dataset argument (i.e. ideally what gets passed to DataLoader)
+        # We want to use the latest possible value for explicit arguments (i.e. ideally what gets passed to base class)
         # so that we can be sure, that it will not get changed anymore.
         # That is why we are setting this in every `__init__`
-        if "dataset" in param_names:
-            setattr(obj, "__dataset", args[param_names.index("dataset")])
-        elif "dataset" in kwargs:
-            setattr(obj, "__dataset", kwargs["dataset"])
+        if store_explicit_args is not None:
+            for explicit_arg in store_explicit_args:
```

Comment:
Is there any occurrence where we make use of the list?

Reply:
No. What happened was that I thought we might need some explicit args for the `BatchSampler` and had no idea how many, so changed it to list and then forgot about it to change it back :) I now made it a single string only, since there is no usecase where the list would be necessary, making it back to the list should not be a big change in the future if we ever need it